### PR TITLE
Add an option to fallback to comments if a description is not provided

### PR DESCRIPTION
**Pull request by rideliner**
## Checklist
- [x] Pull requests follows the [contribution guide](https://github.com/graphql-java-kickstart/graphql-java-tools/wiki/Contribution-guide)
- [x] New or modified functionality is covered by tests

## Description
The existing behavior is to always fallback to comments from the schema. This change would add a configuration option to `SchemaParserOptions` that would make it possible to ignore comments and exclusively use descriptions from the schema.

For example:
```graphql
# Copyright 2020
# Authors: ...

type Query {
    hello: String!
}
```
This schema with the current behavior would result in the description being `Copyright 2020\nAuthors: ...`.
With the `commentsAsFallbackDescription` option set as false, the description will be null.

### CODE REVIEW

File: src/main/kotlin/graphql/kickstart/tools/util/Utils.kt [hunk 1]
```diff
@@ -49,10 +50,17 @@ internal val Class<*>.declaredNonProxyMethods: List<JavaMethod>
...
-internal fun getDocumentation(node: AbstractNode<*>): String? = node.comments?.asSequence()
-    ?.filter { !it.content.startsWith("#") }
-    ?.joinToString("\n") { it.content.trimEnd() }
-    ?.trimIndent()
+internal fun getDocumentation(node: AbstractDescribedNode<*>, options: SchemaParserOptions): String? =
+    when {
+        node.description != null -> node.description.content
+        !options.commentsAsFallbackDescription -> null
+        node.comments == null -> null
+        node.comments.isEmpty() -> null
```
Comment (vojtapol) [inline src/main/kotlin/graphql/kickstart/tools/util/Utils.kt:58]: you can use `isNullOrEmpty()` to combine the two previous lines

File: src/main/kotlin/graphql/kickstart/tools/SchemaParserOptions.kt [hunk 2]
```diff
@@ -30,7 +30,8 @@ data class SchemaParserOptions internal constructor(
     val coroutineContextProvider: CoroutineContextProvider,
     val typeDefinitionFactories: List<TypeDefinitionFactory>,
     val fieldVisibility: GraphqlFieldVisibility?,
-    val includeUnusedTypes: Boolean
+    val includeUnusedTypes: Boolean,
+    val commentsAsFallbackDescription: Boolean
```
Comment (vojtapol) [inline src/main/kotlin/graphql/kickstart/tools/SchemaParserOptions.kt:34]: let's rename this to `useCommentsForDescriptions`

**Review by vojtapol — CHANGES_REQUESTED** [review]

**Comment by oryan-block** [comment]
@rideliner please resolve to conflict. We've migrated all tests to kotlin so you can leave it out. I can recreate your test after that.

File: src/main/kotlin/graphql/kickstart/tools/util/Utils.kt [hunk 3]
```diff
@@ -49,10 +50,16 @@ internal val Class<*>.declaredNonProxyMethods: List<JavaMethod>
...
 
-internal fun getDocumentation(node: AbstractNode<*>): String? = node.comments?.asSequence()
-    ?.filter { !it.content.startsWith("#") }
-    ?.joinToString("\n") { it.content.trimEnd() }
-    ?.trimIndent()
+internal fun getDocumentation(node: AbstractDescribedNode<*>, options: SchemaParserOptions): String? =
+    when {
+        node.description != null -> node.description.content
+        !options.useCommentsForDescriptions -> null
+        node.comments.isNullOrEmpty() -> null
```
Comment (florensie) [inline src/main/kotlin/graphql/kickstart/tools/util/Utils.kt:57]: These two conditions both return `null` so you could just combine them into one.

**Review by oryan-block — APPROVED** [review]
Looks good 🚀

**Review by vojtapol — APPROVED** [review]

**Comment by oryan-block** [comment]
Closing this since we're unable to push/merge it as it is. Opened https://github.com/graphql-java-kickstart/graphql-java-tools/pull/564/ with the same commits.

**Pull request closed without merging**